### ISSUE TITLE
Missing translations

### ISSUE THREAD

**Issue by jobisoft** [jobisoft/TbSync#186]
@sinya13
@ovari
@AlessandroMenti

The master branches of the following projects are ready for the next release:
https://github.com/jobisoft/TbSync
https://github.com/jobisoft/DAV-4-TbSync
https://github.com/jobisoft/EAS-4-TbSync

If you find time, it would help all foreign users if you could help to update/add translations. Thank you very much for your support!

**Comment by sinya13** [comment]
https://github.com/sinya13/TbSync for ru translate - 100% ready
https://github.com/sinya13/CategoryManager for ru translate - 90% ready

**Comment by isorochan** [comment]
Could you give details about the process and rules for adding translations?

**Comment by jobisoft** [comment]
This is the list of the correct quotation marks per language:
https://en.wikipedia.org/wiki/Quotation_mark#Summary_table

**Comment by AlessandroMenti** [comment]
@jobisoft Done in https://github.com/jobisoft/TbSync/pull/188, https://github.com/jobisoft/DAV-4-TbSync/pull/40, https://github.com/jobisoft/EAS-4-TbSync/pull/26.

**Comment by jobisoft** [comment]
All Italien translations merged, thank you @AlessandroMenti

**Comment by sinya13** [comment]
install.rdf also need to be edited like chrome.manifest ?

**Comment by jobisoft** [comment]
Yes, but that is something I can do, but of course, feel free to add yourself. You must add a full description block for your new language and you muist add yourself to all other description blocks.

**Comment by ovari** [comment]
> @ovari Could you provide the link to the list of valid codes? I do not find it anymore.

Sorry, do not understand the context of "list of valid codes". Please email us so this issue does not get filled by this. Thank you

> Could you also provide the link, with the correct quotations marks?

Great you found and listed at https://github.com/jobisoft/TbSync/issues/186#issuecomment-442154462

**Comment by sinya13** [comment]
https://github.com/sinya13/CategoryManager  for ru translate - ready for use.

**Comment by jobisoft** [comment]
@sinya13 You need to create a pull request, which sends your changes back to me:
![image](https://user-images.githubusercontent.com/5830621/49152608-aa2ffd00-f313-11e8-9792-d82b39e4120c.png)

**Comment by sinya13** [comment]
Done.

**Comment by sinya13** [comment]
@isorochan  Можете скопировать русский перевод в папке ../locale/ru/ в папку например ../locale/ua/ или ../locale/ua-UA/ или ../locale/pl/ или ../locale/pl-PL/ в зависимости как в приложении уже начато делаться. Затем во всех строчках меняете русский перевод на державну мову или речь посполиту. Так довольно просто понять что надо менять. По крайней мере проще чем с английской секции править. Затем надо в корне каждого проекта исправит два файла chrome.manifest и install.rdf как это для секций ru там, добавив наалогичную секцию для ua или pl.

**Comment by jobisoft** [comment]
Could you repost that in english?

**Comment by ovari** [comment]
> ../locale/ru/ в папку например ../locale/ua/ или ../locale/ua-UA/ или ../locale/pl/ или ../locale/pl-PL/

@sinya @isorochan not sure what `ua` or `ua-UA` was meant to be; however, folders should be `uk` and `pl` for Ukrainian and Polish, respectively. see [Translations and localizations](https://github.com/jobisoft/TbSync/#translations-and-localizations) → [Transvision](https://transvision.mozfr.org/)

**Comment by sinya13** [comment]
You can copy the russian translation in the folder ../locale/ru/ to a folder such as ../locale/ua/ or ../locale/ua-UA/ or ../locale/pl/ or ../locale/pl-PL / depending on how the application has already begun. Then, in all the lines, change the russian translation to a state or language for ua or pl. So pretty easy to understand what needs to be changed. At least easier than the english section to edit. Then you need to fix two chrome.manifest and install.rdf files in the root of each project like this for the ru sections there, adding a analogical section for ua or pl.

**Comment by sinya13** [comment]
@ovari Yes,probably. Although uk=United Kingdom regular

**Comment by ovari** [comment]
@sinya13 `en-GB`

**Comment by jobisoft** [comment]
@sinya13
@ovari
@AlessandroMenti
@isorochan 

Thanks for all your work. The new UI is almost ready. I am currently adding a server profile selection (when creating accounts, the user can pick from pre-defined servers without having to enter any URL) and I needed to updated a few things to comply with Thunderbird rules. The current locale files include these changes already.

It is not much, could you take a look at:

https://github.com/jobisoft/TbSync
https://github.com/jobisoft/DAV-4-TbSync
https://github.com/jobisoft/EAS-4-TbSync

and update any translation embraced by "??" like
```
<!ENTITY add.serverprofile "?? Select a server profile: ??">
``` 

@ovari Do you plan to translate the DAV provider as well? The UX would improve a lot, because DAV users would have a translated TbSync manager, but the DAV settings window would not be translated...

As soon as I have these final translations, I will release beta versions for you to try out. This is how the new UI will look like:

![image](https://user-images.githubusercontent.com/5830621/49505662-88d39180-f87c-11e8-8c07-922b0cb7a515.png)

![image](https://user-images.githubusercontent.com/5830621/49505691-9c7ef800-f87c-11e8-81f9-b53dcb591d1f.png)

![image](https://user-images.githubusercontent.com/5830621/49505707-a143ac00-f87c-11e8-93b8-7578f968d400.png)

![image](https://user-images.githubusercontent.com/5830621/49505730-a86aba00-f87c-11e8-9f7b-f08bb6c58ed4.png)

**Comment by jobisoft** [comment]
I just released betas for you to try out. See your translations in action!

https://tbsync.jobisoft.de/beta/TbSync.xpi
https://tbsync.jobisoft.de/beta/DAV-4-TbSync.xpi
https://tbsync.jobisoft.de/beta/EAS-4-TbSync.xpi

I will send this to Thunderbird review in 2-3 days.

**Comment by sinya13** [comment]
At first glance - good!
And will CategoryManager be collected with russification?

**Comment by jobisoft** [comment]
CatMan will also be send to Thunderbird this week.

Best regards,
John

**Comment by sinya13** [comment]
Tell me a link to this assembly CategoryManager please. But I do not find it with the translation.And I can not see what happened in Thunderbird.

**Comment by jobisoft** [comment]
Can you rephrase that? I did not understand, sorry. If you have more questions on CatMan, let's open a new issue.

**Comment by sinya13** [comment]
I wanted http/https link to download last  beta bundle of CategoryManager with ru-translation.

**Comment by jobisoft** [comment]
Understood! Will pack one for you this afternoon.

**Comment by jobisoft** [comment]
@sinya13 Created a dedicated issue for CatMan, where you can get the latest beta incl. your translations:
https://github.com/jobisoft/CategoryManager/issues/24

**Comment by sinya13** [comment]
Thanks!

**Comment by sinya13** [comment]
There are small wishes for the program part:

1.The original width of the main window TbSynk is not enough 1-2 centimeters. The right edge of the "synchronize now" button (in ru "синхронизировать сейчас") and the ends of mail addresses in "Status" of the right panel do not fit.

2. The boundaries of the table columns are rigidly fixed horizontally and they cannot be shifted to view non-fit parameters. Is it possible to make them movable?

**Comment by jobisoft** [comment]
@ all: First of all, if you feel you get too many emails from this issue, I am deeply sorry. But everything that is being discussed here is indeed important to me and crucial for the next release. So if you think most of it is "off topic" for your, I kindly ask you to unsubscribe from this issue. They will probably be a lot more emails coming.

@sinya13 Thanks for testing the new version with your locales. As of TB60 it is no longer possible for me to switch locales, but I would have to actually install a Russian version of TB to see your locales. Thanks for testing! 

Lets handle your findings one-by-one:
![image](https://user-images.githubusercontent.com/5830621/49867923-33fbc200-fe0c-11e8-91b3-a56a260cc9ab.png)

This is odd. I am using standard tab element and I am not doing any styling to them. They should look like this:

![image](https://user-images.githubusercontent.com/5830621/49867986-5ab9f880-fe0c-11e8-9acb-24ba2b13586c.png)

Do you have any special theme installed? What OS are you working on? I do not think I can do anything about that :-(

![image](https://user-images.githubusercontent.com/5830621/49868033-79b88a80-fe0c-11e8-84a5-e9c38843371c.png)

Can you shorten the locales involved here? They are (in sum) too long so they push the button out of the designated area. If possible, I would not like to allow the "sync interval" locale to break into two lines. Any chance you can rephrase the locale to make it shorter?

![image](https://user-images.githubusercontent.com/5830621/49868197-f2b7e200-fe0c-11e8-898a-354847520a12.png)

I do not know how to make these movable. You could simply use a shorter name for your account (its is realy just a name, your username is stored elsewhere)

![image](https://user-images.githubusercontent.com/5830621/49868245-2266ea00-fe0d-11e8-89a2-5e114ed16780.png)

Again, I cannot make them movable. But I am currently reworking that part:

![image](https://user-images.githubusercontent.com/5830621/49868281-3f032200-fe0d-11e8-8cf7-0111107ba220.png)

I will allow line breaks here so all information is visible. I will push that over the beta release channel in 1 or 2 days (Which means the final release to Thunderbird is postponed once again).

![image](https://user-images.githubusercontent.com/5830621/49868350-7d004600-fe0d-11e8-9f4e-27c70037efae.png)

I will have to remove the Category Manager tab altogether, I think. Furthermore, it does not fit in that region where you moved it. That is supposed to be the area, where contributors like you should be mentioned, but it is not intuitive enough: You must click on the icon of TbSync or one of the providers to open the extra window. I will add dedicated buttons there to improve UX.

One final note: Please uninstall the EWS provider. I do not keep its API calls up to date so it could break your TbSync. It ist still a looooong way until the EWS provider will be usable.

**Comment by sinya13** [comment]
Ok. Only wishes! Everything works - this is the main thing! 
"Синхронизировать сейчас" = "Синхронизировать" only.
My configuration: 
Linux Ubuntu x64 16.04 XFCE
Thunderbird 60.2.1 theme = standart

**Comment by sinya13** [comment]
What a simple, shortly language for Kerio makes this setting:
<!ENTITY pref.provision "Enforce provisioning (required by Kerio)"> ?

The phrase "Enforce provisioning" is translated clumsily.
Does it duplicate requests to server type Kerio?

**Comment by jobisoft** [comment]
Kerio Servers return an error 500, if that Option is not set. That is a bug in kerio, it should return errror 449 (or so) and tbsync would than activate that Option by itself.

**Comment by jobisoft** [comment]
the technical term is "provisioning"

**Comment by sinya13** [comment]
@jobisoft Ok. In russian "provisioning" term from grocery area - type "bring for example food" - provisioning food - clumsily.

**Comment by jobisoft** [comment]
I will check tonight, if Kerio fixed the bug, which would render the Option obsolete.

In English "provisioning" is more like "making something available". If it is not translatable, I would use 

```
Enforce "EAS Provison" (required by Kerio)
``` 

and not translate the technical term at all?

**Comment by sinya13** [comment]
@jobisoft The user is not always a technical person. Sometimes not canonical, but a simple translation is better. But to translate the names of sites and the name of programs - do not ever! My opinion.

**Comment by sinya13** [comment]
@jobisoft "sync interval"= "интервал синхронизации"="период синхронизации"
Good?

**Comment by jobisoft** [comment]
Hm, other translations might be too long as well. I think I do need to allow a line break to fix this. Stay tuned. Do not change locales there.

**Comment by jobisoft** [comment]
@sinya13 
I released new betas. Can you update the add-ons via "check for updates" and take a look?

**Comment by sinya13** [comment]
@jobisoft Unfortunately I probably won't be able to do it until Monday...

**Comment by sinya13** [comment]
![tbsync-m-1-1](https://user-images.githubusercontent.com/4994987/49920584-c1e2b600-febb-11e8-8178-2fb25b0fa23a.jpg)

**Comment by jobisoft** [comment]
The thing in the lower left is a button element, the stuff at the top is a tabpanel element. I do not know, why the tabpanel element is rendered wrong on your system :-( 

I cannot replace the tabpanel element with buttons :-(

**Comment by sinya13** [comment]
@jobisoft
A button is forbidden to put there? Can it be only in the corners?

**Comment by jobisoft** [comment]
No, not forbidden :-) But conceptional I need panels there

**Comment by jobisoft** [comment]
This has something to do wit your OS, I think:
https://github.com/MoonchildProductions/UXP/issues/777

**Comment by sinya13** [comment]
Well, in principle, it works the same way ... Maybe the feature of simplified XFCE ... And for example, everything is fine on KDE. But XFCE is popular in Linux ...

**Comment by sinya13** [comment]
@jobisoft
And so? Is this similar item in there? Not an option? :-)
![tbsync-m-1-1](https://user-images.githubusercontent.com/4994987/49921765-aa0d3100-febf-11e8-89d0-06612415d2e6.jpg)

**Comment by sinya13** [comment]
@jobisoft
I don't have a DAV server working, but I see one algorithm error! It can only be corrected by restarting Thunderbird only !!!
![dav-err-1](https://user-images.githubusercontent.com/4994987/49937291-557dac00-fee7-11e8-8e3e-72386ec44f20.jpg)

**Comment by jobisoft** [comment]
Please send in a bug report (at the "help" tap) . This is due to some JavaScript error.

**Comment by sinya13** [comment]
@jobisoft "Cannot find valid debug messages". No errors - just the counter continues to count to a minus indefinitely.

**Comment by jobisoft** [comment]
- please go to the help tap, 
- activate debug logging, 
- restart TB, 
- retrigger whatever you did to get that negative counter, 
- if the counter is negative, go the help tap
- create a bug report

**Comment by jobisoft** [comment]
Thanks, nice litte bug you found there...

**Comment by jobisoft** [comment]
fixed? Update addons and try again.

As a side note: The instructions given at https://github.com/jobisoft/TbSync/issues/186#issuecomment-446949234 should also have been given to you via a popup after clicking on "create bug report" while no debug data avail. Did that not work for you?

**Comment by sinya13** [comment]
"For this server could not find the service saber/dav" -  message. The counter does not appear any more.  I forgot that need to restart thunderbird for the report to appear ...
